fix(ui): framer-motion removed CustomDomComponent — fix elements/animated.ts

From triaging Renovate #33 (framer-motion 11.18.2, minor) and #90 (v12, major) — **same root cause blocks both**:

```
elements/animated.ts(5,15): error TS2305: Module '"framer-motion"' has no exported member 'CustomDomComponent'.
components/header/title.tsx(126,6): error TS2322 (cascade from the broken motionChakra return type)
```

`CustomDomComponent` was removed from framer-motion's public types. `elements/animated.ts` uses it as the return type of the `motionChakra()` factory; `title.tsx`'s error is downstream cascade.

**Fix (1–2 files):** replace the annotation with a v11.18+/v12-compatible type (e.g. `DOMMotionComponents`-derived or drop the explicit annotation and let inference work — the existing `// @ts-expect-error` on the merge call already flags this type as fragile).

**Unblocks:** #33 (minor → automerges once green) and #90 (major — independent of the chakra v3 epic, since chakra v3 no longer bundles framer-motion).

--- a/hyperglass/ui/components/header/title.tsx
+++ b/hyperglass/ui/components/header/title.tsx
@@ -1,8 +1,8 @@
 import { Button, Flex, VStack } from '@chakra-ui/react';
-import { motion } from 'framer-motion';
 import { isSafari } from 'react-device-detect';
 import { Case, Switch } from 'react-if';
 import { useConfig } from '~/context';
+import { motionChakra } from '~/elements';
 import { useFormInteractive, useFormState, useMobile } from '~/hooks';
 import { Logo } from './logo';
 import { SubtitleOnly } from './subtitle-only';
@@ -15,8 +15,10 @@ type DWrapperProps = Omit<StackProps, 'transition'> & MotionProps;
 type MWrapperProps = Omit<StackProps, 'transition'> & MotionProps;
 type WrapperProps = Partial<MotionProps & Omit<StackProps, 'transition'>>;
 
-const AnimatedVStack = motion(VStack);
-const AnimatedFlex = motion(Flex);
+// motionChakra (rather than raw `motion(...)`) reconciles the conflicting
+// chakra/framer `transition` prop types — see elements/animated.ts.
+const AnimatedVStack = motionChakra<StackProps>(VStack);
+const AnimatedFlex = motionChakra<FlexProps>(Flex);
 
 /**
  * Title wrapper for mobile devices, breakpoints sm & md.
@@ -115,7 +117,10 @@ const All = (props: WrapperProps): JSX.Element => (
  * Title component which renders sub-components based on the `title_mode` configuration variable.
  */
 export const Title = (props: FlexProps): JSX.Element => {
-  const { fontSize, ...rest } = props;
+  // framer-motion and chakra/DOM declare incompatible types for these props;
+  // AnimatedFlex expects framer's versions, so exclude the DOM ones from the
+  // spread (they are never passed — <Title /> takes no props at its call site).
+  const { fontSize, transition, onAnimationStart, onDragStart, onDragEnd, onDrag, ...rest } = props;
   const { web } = useConfig();
   const { titleMode } = web.text;
 

--- a/hyperglass/ui/elements/animated.ts
+++ b/hyperglass/ui/elements/animated.ts
@@ -2,7 +2,7 @@ import { chakra } from '@chakra-ui/react';
 import { motion } from 'framer-motion';
 
 import type { BoxProps } from '@chakra-ui/react';
-import type { CustomDomComponent, Transition, MotionProps } from 'framer-motion';
+import type { MotionProps, Transition } from 'framer-motion';
 
 type MCComponent = Parameters<typeof chakra>[0];
 type MCOptions = Parameters<typeof chakra>[1];
@@ -11,7 +11,21 @@ type MakeMotionProps<P extends BoxProps> = React.PropsWithChildren<
 >;
 
 /**
+ * Return type of `motionChakra`. Self-defined replacement for framer-motion's
+ * `CustomDomComponent` (removed from the public types in v11.x), mirroring its
+ * shape so the signature survives framer-motion type churn.
+ */
+type MotionChakraComponent<P extends BoxProps> = React.ForwardRefExoticComponent<
+  MakeMotionProps<P> & React.RefAttributes<HTMLElement>
+>;
+
+/**
  * Combine `chakra` and `motion` factories.
+ *
+ * Chakra and framer-motion both declare a `transition` prop (CSS string vs.
+ * animation config), so the combined component's props are re-declared with
+ * chakra's omitted — the cast below is deliberate and is what makes the
+ * merged component usable without per-call-site suppressions.
  *
  * @param component Component or string
  * @param options `chakra` options
@@ -20,9 +34,8 @@ type MakeMotionProps<P extends BoxProps> = React.PropsWithChildren<
 export function motionChakra<P extends BoxProps = BoxProps>(
   component: MCComponent,
   options?: MCOptions,
-): CustomDomComponent<MakeMotionProps<P>> {
-  // @ts-expect-error I don't know how to fix this.
-  return motion<P>(chakra<MCComponent, P>(component, options));
+): MotionChakraComponent<P> {
+  return motion(chakra(component, options)) as unknown as MotionChakraComponent<P>;
 }
 
 export const AnimatedDiv = motionChakra('div');

